CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/RawSyntaxValidationFile.swift

```diff
@@ -0,0 +1,176 @@
+                      Node did not satisfy any node choice requirement.
+                      Validation failures:
+                      \(nonNilErrors.map({ "- \($0.description)" }).joined(separator: "\n"))
+                      """, file: file, line: line)
+                    _ = 1
+                  }
+                }
+                """#
+              )
+
+              try! SwitchExprSyntax("switch kind") {
+                SwitchCaseSyntax(
+                  """
+                  case .token:
+                    assertionFailure("validateLayout for .token kind is not supported")
+                  """
+                )
+
+                for node in NON_BASE_SYNTAX_NODES {
+                  SwitchCaseSyntax("case .\(raw: node.swiftSyntaxKind):") {
+                    if node.isBuildable || node.isMissing {
+                      ExprSyntax("assert(layout.count == \(raw: node.children.count))")
+                      for (index, child) in node.children.enumerated() {
+                        switch child.kind {
+                        case .nodeChoices(let choices):
+                          let verifiedChoices = ArrayExprSyntax {
+                            ArrayElementSyntax(
+                              leadingTrivia: .newline,
+                              expression: ExprSyntax("verify(layout[\(raw: index)], as: Raw\(raw: child.type.buildable).self)")
+                            )
+                          }
+
+                          ExprSyntax("assertAnyHasNoError(kind, \(raw: index), \(verifiedChoices))")
+                        default:
+                          ExprSyntax("assertNoError(kind, \(raw: index), verify(layout[\(raw: index)], as: Raw\(raw: child.type.buildable).self))")
+                        }
+                      }
+                    } else if node.isSyntaxCollection {
+                      try! ForInStmtSyntax("for (index, element) in layout.enumerated()") {
+                        if let collectionElementChoices = node.collectionElementChoices, !collectionElementChoices.isEmpty {
+                          let verifiedChoices = ArrayExprSyntax {
+                            for choiceName in node.collectionElementChoices! {
+                              let choice = SYNTAX_NODE_MAP[choiceName]!
+                              ArrayElementSyntax(
+                                leadingTrivia: .newline,
+                                expression: ExprSyntax("verify(element, as: Raw\(raw: choice.name).self)")
+                              )
+                            }
+                          }
+                          ExprSyntax("assertAnyHasNoError(kind, index, \(verifiedChoices))")
+                        } else {
+                          ExprSyntax("assertNoError(kind, index, verify(element, as: Raw\(node.collectionElementType.buildable).self))")
+                        }
+                      }
+                    }
+
+                    BreakStmtSyntax()
```

Comment:
Again, not part of this PR but I don’t think we need the `break`. It’s probably a leftover of someone writing too much C++ before writing `RawSyntaxValidation.swift`